include/util/json_renderer.hpp

```diff
@@ -27,18 +27,32 @@ namespace
 constexpr int MAX_FLOAT_STRING_LENGTH = 256;
 }
 
-struct Renderer
+template <typename Out> struct Renderer
 {
-    explicit Renderer(std::ostream &_out) : out(_out) {}
+    explicit Renderer(Out &_out) : out(_out) {}
 
-    void operator()(const String &string) const
+    void operator()(const String &string)
     {
-        out << "\"";
-        out << escape_JSON(string.value);
-        out << "\"";
+        write('"');
```

Comment:
# 2.
Then I noticed that we spend quite a lot of time on escaping the strings which usually don't need to be escaped:
<img width="1183" alt="precalculate escapeed string size" src="https://user-images.githubusercontent.com/266271/193302126-925d6467-880a-45e2-b13f-85bbb4c4e6a4.png">